connaisseur/flask_server.py

```diff
@@ -135,35 +134,30 @@ def __create_logging_msg(msg: str, **kwargs):
     return str({"message": msg, "context": dict(**kwargs)})
 
 
-def __create_json_patch(path: str, index: int, image: Image):
-    return {"op": "replace", "path": path.format(str(index)), "value": str(image)}
-
-
 def __admit(admission_request: AdmissionRequest):
-    policy = ImagePolicy()
-    parent_images = admission_request.wl_object.parent_images
-    patches = []
+    try:
+        logging_context = dict(admission_request.context)
+        policy = ImagePolicy()
+        patches = []
 
-    for index, c_image in enumerate(admission_request.wl_object.container_images):
-        try:
-            logging_context = dict(admission_request.context, image=c_image)
+        for container in admission_request.wl_object.containers:
+            logging_context.update(image=str(container.image))
 
             # child resources have mutated image names, as their parents got mutated
             # before their creation. this may result in mismatch of rules or duplicate
             # lookups for already approved images. so child resources are automatically
             # approved without further check ups, when their parents were approved
             # earlier.
-            if c_image in parent_images:
-                msg = f'automatic child approval for "{c_image}".'
+            if container in admission_request.wl_object.parent_containers:
```

Comment:
correct me if I'm wrong, but if you use sth like
`if container.image in [container['image'] for container in admission_request.wl_object.parent_containers]:` here instead you could get rid of the controversial `__eq__` function for containers, couldn't you?

Reply:
did go for this implementaion: https://github.com/sse-secure-systems/connaisseur/pull/157#discussion_r666315210